openmeetings-db/src/main/java/org/apache/openmeetings/db/entity/file/BaseFileItem.java

```diff
@@ -296,7 +297,31 @@ public String getExternalType() {
 	public void setExternalType(String externalType) {
 		this.externalType = externalType;
 	}
-
+	
+	/**
+	 * Generates a link to download. As opposed to display it on the Whiteboard.
+	 * 
+	 * @param ext
+	 * @return
+	 */
+	public final File getDownloadFile(String ext) {
```

Comment:
This method is not necessary

Reply:
I don't think so. Neither getOriginal or getFile(null) is a replacement for it.

getOriginal: We don't _want to download the original_ !
we want to download in case of a PPT, the ppt OR the PDF (if you go for the drop down menu). Depending on what you pass in as "ext". But passing in null does just not work.

You are right that this logic could be moved into the FileTreePanel and it passes in the right ext instead of passing in null. But that is also a bit arbitrary.
Cause the logic as it also was implemented assumes 'if you pass in null it returns the original', which is just not true.

Reply:
> getOriginal: We don't want to download the original !
Well the menu item has name `Original` and it was intended to download original :)))
So this is the method which in charge here :)

Reply:
well neither does the button trigger that method nor does the method actually return the original.